File: leap_year/erlang_leap_year/leap_year.erl
% comment
-module(leap_year).
%-export([start/0]).
-compile(export_all).

is_leap_year(Year) when Year rem 400 =:= 0 -> true;
is_leap_year(Year) when Year rem 100 =:= 0 -> false;
is_leap_year(Year) when Year rem   4 =:= 0 -> true;
is_leap_year(_)                            -> false.

for_loop(X) when X == 0 -> 0;
for_loop(X) when X > 0 -> for_loop(X-1), io:fwrite("is_leap_year(~w)=<~w>~n",[X,is_leap_year(X)]).

start() ->
  io:fwrite(" *** need to fix *** ~n"),
  for_loop(10),
  io:fwrite(" *** need to fix *** ~n").

File: leap_year/typescript_leap_year/leap_year.ts
//comment
var isLeapYear = function (year: number) {
  if( year > 1581 ) {
    return(((year % 4 == 0) && (year % 100) > 0) || (year % 400 == 0));
  } else {
    return false;
  }
};

var main = function(){
  for(var idx_i = 1999; idx_i < 2101; idx_i++) {
    console.log("isLeapYear(" + idx_i + ")=<" + isLeapYear(idx_i) + ">");
  }
};

main();
